<launch>

  <!--<arg name="bag" default="vicon_fix/gareth1_2014-12-15-18-59-45"/>-->
  <arg name="bag" default="statefarm/small"/>
  <!--<arg name="bag" default="vicon_fix/gareth6_2014-12-15-19-04-46"/>-->
  <arg name="bag_rate" value="1"/>
  <arg name="bag_start" value="65"/>

  <!-- plays the bag file -->
  <include file="$(find galt_setup)/launch/process/play_bag.launch">
    <arg name="bag" value="$(arg bag)"/>
    <arg name="bag_rate" value="$(arg bag_rate)"/>
    <arg name="bag_start" value="$(arg bag_start)"/>
  </include>

  <!-- publishes TF -->
  <include file="$(find galt_setup)/launch/process/process_base.launch">
    <arg name="profile" value="steadicam/v4"/>
    <arg name="vicon" value="true"/>
  </include>

  <!-- camera to use for odometry -->
  <arg name="camera_name" value="/stereo/left"/>

  <node pkg="svo_ros" type="vo" name="svo"
    clear_params="true" output="screen">

    <param name="publish_dense_input" type="bool" value="true"/>
    <param name="publish_markers" type="bool" value="true"/>

    <remap from="~image" to="$(arg camera_name)/image_raw"/>
    <remap from="~camera_info" to="$(arg camera_name)/camera_info"/>

    <!-- Default parameter settings: choose between vo_fast and vo_accurate -->
    <rosparam file="$(find svo_ros)/param/vo_accurate.yaml" />
  </node>
</launch>
